src/state.rs

```diff
@@ -763,6 +814,14 @@ impl DemandProxyState {
     }
 }
 
+#[derive(Eq, PartialEq, Clone, Copy, Debug)]
+pub enum ServiceResolutionMode {
```

Comment:
Comment here explaining why we need to modes and what we use this for would be good.

(I see why but in 6 months local context will help)